OneCommunityGlobal/HighestGoodNetworkApp
# Zubing - Add chatgpt write it for me button

# Description
<img width="744" alt="Screenshot 2023-11-18 at 6 25 30 AM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/92126559/5c105f15-442b-49e3-afa3-479608046b3a">

## Related PRS (if any):
This frontend PR is related to the backend PR. https://github.com/OneCommunityGlobal/HGNRest/pull/646
To test this frontend PR you need to checkout the #646 backend PR.

## Main changes explained:
- Update WeeklySummary.jsx for including the WriteItForMe button 
- Create WriteIteForMeModal.jsx for introducing new components Write It For Me

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as a user with some tangible time entries in this week
4. go to WeeklySummary submitting page
5. click the "Write It For Me" button
6. check the popup window and displaying of the summary
7. click "Copy and Paste", check if it is copied.

## Screenshots or videos of changes:

## Note:
The issue is : When running npm test, the test of WeeklySummary.test.jsx failed. You can scroll down to see the failing details.
![Fail summary](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/92126559/a5eb276a-900c-4596-973e-eeea5c969050)

The WeeklySummary component requires access to the Redux store, it can be resolved  by wrapping the component with a <Provider> from react-redux . The way to do that is ( I think) : in WeeklySummary.test.jsx,  create a mock store and wrap the component with a <Provider> that passes this store. I tried to code for it, but it didn’t really work.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/92126559/4faf92c7-a52b-48dd-bc3b-157ce34ca79e
